pkg/controller/application/application_controller_utils.go

```diff
@@ -7,47 +7,38 @@ import (
 	"strconv"
 
 	"github.com/golang/glog"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 
 	shipperv1 "github.com/bookingcom/shipper/pkg/apis/shipper/v1"
 	"github.com/bookingcom/shipper/pkg/controller"
+	"github.com/bookingcom/shipper/pkg/errors"
 )
 
-func (c *Controller) createReleaseForApplication(app *shipperv1.Application, generation int) error {
+func (c *Controller) createReleaseForApplication(app *shipperv1.Application, releaseName string, iteration, generation int) (*shipperv1.Release, error) {
 	// Label releases with their hash; select by that label and increment if needed
 	// appname-hash-of-template-iteration.
-	releaseName, iteration, err := c.releaseNameForApplication(app)
-	if err != nil {
-		return err
-	}
-	releaseNs := app.GetNamespace()
-	glog.V(4).Infof("Generated Release name for Application %q: %q", controller.MetaKey(app), releaseName)
-
-	labels := make(map[string]string)
-	for k, v := range app.GetLabels() {
-		labels[k] = v
-	}
-
-	labels[shipperv1.ReleaseLabel] = releaseName
-	labels[shipperv1.AppLabel] = app.GetName()
-	labels[shipperv1.ReleaseEnvironmentHashLabel] = hashReleaseEnvironment(app.Spec.Template)
+	var (
```

Comment:
nit: I think this can be nuked, the only 'err' in the function is := defined later on.